apps/system/js/screen_auto_brightness.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+(function(exports) {
+
+  /*
+   * constructor - argument is the function that will adjust
+   * the screen brightness. This function takes 2 arguments,
+   * the brightness and whether the change is immediate (true),
+   * or will be transitioned (false)
+   */
+  var ScreenAutoBrightness = function ScreenAutoBrightness(
+      setScreenBrightnessFn) {
+    this._setScreenBrightness = setScreenBrightnessFn;
```

Comment:
It's not a convention (at least for me personally) to pass a callback function.

I would do

```javascript
ScreenAutoBrightness.prototype.onbrightnesschange = null;
```

and in the calling site, do

```javascript
if (this.onbrightnesschange === 'function') {
  this.onbrightnesschange(clampedBrightness);
} else {
  console.error('ScreenAutoBrightness: Brightness should change but no callback attached.');
}
```

and attach the callback in ScreenManager with

```javascript
this._screenAutoBrightness.onbrightnesschange = function(brightness) {
  this.setScreenBrightness(brightness, false);
}.bind(this);
```